transfuse-core/src/main/java/org/androidtransfuse/util/apache/commons/WriterOutputStream.java

```diff
@@ -0,0 +1,146 @@
+        while(len > 0) {
+            int c = Math.min(len, this.decoderIn.remaining());
+            this.decoderIn.put(b, off, c);
+            this.processInput(false);
+            len -= c;
+            off += c;
+        }
+
+        if (this.writeImmediately) {
+            this.flushOutput();
+        }
+
+    }
+
+    public void write(byte[] b) throws IOException {
+        this.write(b, 0, b.length);
+    }
+
+    public void write(int b) throws IOException {
+        this.write(new byte[]{(byte)b}, 0, 1);
+    }
+
+    public void flush() throws IOException {
+        this.flushOutput();
+        this.writer.flush();
+    }
+
+    public void close() throws IOException {
+        this.processInput(true);
+        this.flushOutput();
+        this.writer.close();
+    }
+
+    private void processInput(boolean endOfInput) throws IOException {
+        this.decoderIn.flip();
+
+        while(true) {
+            CoderResult coderResult = this.decoder.decode(this.decoderIn, this.decoderOut, endOfInput);
+            if (!coderResult.isOverflow()) {
+                if (coderResult.isUnderflow()) {
+                    this.decoderIn.compact();
+                    return;
+                } else {
+                    throw new IOException("Unexpected coder result");
+                }
+            }
+
+            this.flushOutput();
+        }
+    }
+
+    private void flushOutput() throws IOException {
+        if (this.decoderOut.position() > 0) {
+            this.writer.write(this.decoderOut.array(), 0, this.decoderOut.position());
+            this.decoderOut.rewind();
+        }
+
+    }
+
+    private static void checkIbmJdkWithBrokenUTF16(Charset charset) {
+        if ("UTF-16".equals(charset.name())) {
+            String TEST_STRING_2 = "vés";
+            byte[] bytes = "vés".getBytes(charset);
+            CharsetDecoder charsetDecoder2 = charset.newDecoder();
+            ByteBuffer bb2 = ByteBuffer.allocate(16);
+            CharBuffer cb2 = CharBuffer.allocate("vés".length());
+            int len = bytes.length;
+
+            for(int i = 0; i < len; ++i) {
+                bb2.put(bytes[i]);
+                bb2.flip();
+
+                try {
+                    charsetDecoder2.decode(bb2, cb2, i == len - 1);
+                } catch (IllegalArgumentException var9) {
+                    throw new UnsupportedOperationException("UTF-16 requested when runninng on an IBM JDK with broken UTF-16 support. Please find a JDK that supports UTF-16 if you intend to use UF-16 with WriterOutputStream");
+                }
+
+                bb2.compact();
+            }
+
+            cb2.rewind();
+            if (!"vés".equals(cb2.toString())) {
+                throw new UnsupportedOperationException("UTF-16 requested when runninng on an IBM JDK with broken UTF-16 support. Please find a JDK that supports UTF-16 if you intend to use UF-16 with WriterOutputStream");
+            }
+        }
+    }
+}
```

Comment:
Go ahead and get rid of anything not used.  Since we're using utf-8, the `checkIbmJdkWithBrokenUTF16` will never execute, and most of the constructors can be combined into one.

Reply:
done, i just removed it and combine the constructor.